common/http.go

```diff
@@ -302,25 +302,29 @@ func (h *HTTPClient) httpGet(ctx context.Context, //nolint:dupl
 
 // httpPost makes a POST request to the given URL and returns the response & response body.
 func (h *HTTPClient) httpPost(ctx context.Context, url string, //nolint:dupl
-	headers []Header, body []byte,
+	headers []Header, requestBody []byte,
 ) (*http.Response, []byte, error) {
-	req, err := makePostRequest(ctx, url, headers, body)
+	req, err := makePostRequest(ctx, url, headers, requestBody)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	correlationId := uuid.Must(uuid.NewRandom()).String()
 
 	if logging.IsVerboseLogging(ctx) {
-		logRequestWithBody(logging.VerboseLogger(ctx), req, "POST", correlationId, url, body)
+		// body is nil because makePostRequest sometimes returns an altered body reader
+		// so we rely on req and not body for logging purposes. If we use body, it will
+		// not just log the wrong thing, it will swap the body out, which will lead to
+		// unexpected behavior in the request.
+		logRequestWithBody(logging.VerboseLogger(ctx), req, "POST", correlationId, url, nil)
```

Comment:
@eberle1080 would it be better to modify the signature of this function to not have the last parameter?